btcstaking/types.go

```diff
@@ -267,6 +270,38 @@ type babylonScriptPaths struct {
 	slashingPathScript []byte
 }
 
+func keyToString(key *btcec.PublicKey) string {
+	return hex.EncodeToString(schnorr.SerializePubKey(key))
+}
+
+func checkForDuplicateKeys(
+	stakerKey *btcec.PublicKey,
+	fpKeys []*btcec.PublicKey,
+	covenantKeys []*btcec.PublicKey,
+) error {
+	keyMap := make(map[string]bool)
```

Comment:
```suggestion
	keyMap := make(map[string]struct{})
```

is better, see https://www.reddit.com/r/golang/comments/qzha6o/for_sets_maptstruct_vs_maptbool/

Reply:
it is better only if you care about memory ie the map is predicted to be large, here I would say it is not the case and imo version with bools is a bit more readable.

Saying all that, it is not that strong opinion, so I will change it 👍